python/Ganga/Lib/LCG/Grid.py

```diff
@@ -658,14 +658,14 @@ def get_output(self, jobid, directory, wms_proxy=False):
             logger.warning('GRID proxy lifetime shorter than 1 hour')
             return (False, None)
 
-        cmd = '%s --noint --dir %s %s' % (cmd, directory, jobid)
+        cmd = '%s --noint --dir "%s" %s' % (cmd, directory, jobid)
 
         logger.debug('job get output command: %s' % cmd)
 
         rc, output, m = self.shell.cmd1('%s%s' % (
             self.__get_cmd_prefix_hack__(binary=exec_bin), cmd), allowed_exit=[0, 255])
 
-        match = re.search('directory:\n\s*(\S+)\s*\n', output)
+        match = re.search('directory:\n\s*(.+)\s*\n', output)
```

Comment:
I guess this becomes the tricky question of "what is a valid filename in Linux?". I suppose it can't hurt to be a little more explicit about it though as those characters you listed are realistically not being to be used in file names.